CRM/Mailing/BAO/Mailing.php

```diff
@@ -357,22 +357,40 @@ public static function getRecipients($mailingID) {
       $query = $query->where($aclWhere);
     }
 
+    // TODO: read the directory from the MySQL server
+    $tmpfname = self::tmpfile_for_select_into("/var/lib/mysql-files",$mailingID);
+
     // this mean if dedupe_email AND the mysql 5.7 supports ONLY_FULL_GROUP_BY mode then as
     //  SELECT must contain 'email' column as its used in GROUP BY, so in order to resolve This
     //  here the whole SQL code is wrapped up in FROM table i and not selecting email column for INSERT
     if ($key = array_search('e.email', $selectClause)) {
       unset($selectClause[$key]);
       $sql = $query->toSQL();
-      CRM_Utils_SQL_Select::from("( $sql ) AS i ")
+      $sql = CRM_Utils_SQL_Select::from("( $sql ) AS i ")
         ->select($selectClause)
-        ->insertInto('civicrm_mailing_recipients', ['mailing_id', 'contact_id', $entityColumn])
         ->param('#mailingID', $mailingID)
-        ->execute();
-    }
-    else {
-      $query->insertInto('civicrm_mailing_recipients', ['mailing_id', 'contact_id', $entityColumn])
+        ->toSQL();
+
+      // Error check?
```

Comment:
I imagine an error would throw an exception, to be handled downstack, so my guess would be no error check. But maybe worth a test.